### PR TITLE
Alert control

### LINKED ISSUE #25: Alert
Overview:
Display direct response to a user action. They appear at the top of the content area
Doc:
https://modus.trimble.com/components/mobile/alerts/

Specifications:
- Container
- Alert text
- Leading Non-Interactive Icon/ Charm 
- Button (optional)

### PR DESCRIPTION
**Pull request by BAL4-R**
Added Alert Control, closes #25

Simulator Screenshot - iPhone 15 - 2023-11-24 at 10 00 49
simulator_screenshot_ED126E00-CDC4-4CB2-98D8-6A61EBCE8F25

### CODE REVIEW

**Review by rthanga1 — APPROVED** [review]

**Review by ElishaSamPeterPrabhu — CHANGES_REQUESTED** [review]
Background Color Is being applied according to dark theme but the icons are different from figma
Success :
image
Error :
image
Warning :
image
Primary :
image
Secondary :
image
Dark :
image
Expected :
image

File: DemoApp/DemoApp/Views/AlertSamplePage.xaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      </HorizontalStackLayout>
+      <HorizontalStackLayout VerticalOptions="Center"
+                             Margin="5">
+        <Label Text="Dismissable"
+                VerticalOptions="Center"
+                Margin="0,0,5,0" />
+        <Switch IsToggled="{Binding Dismissable}"
+                VerticalOptions="Center" />
+      </HorizontalStackLayout>
+      <Button Text="Show"
```
Comment (rthanga1) [inline DemoApp/DemoApp/Views/AlertSamplePage.xaml:47]: Use the modus button and modus switches in this page?

File: Trimble.Modus.Components/Controls/TMAlert/TMAlert.xaml [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+            HorizontalOptions="FillAndExpand"
+            ColumnSpacing="0"
+            Margin="0"
+            Padding="0">
+        <Grid.ColumnDefinitions>
+          <ColumnDefinition Width="Auto"/>
+          <ColumnDefinition Width="*"/>
+        </Grid.ColumnDefinitions>
+
+        <BoxView x:Name="VerticalSeparator"
```
Comment (rthanga1) [inline Trimble.Modus.Components/Controls/TMAlert/TMAlert.xaml:45]: If you look closely there's a spacing in left and top of the separator.

Screen Shot 2023-12-01 at 3 56 51 PM

File: Trimble.Modus.Components/Controls/TMAlert/TMAlert.xaml [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+            BindingContext="{x:Reference alertView}"
+            StrokeThickness="1"
+            Margin="0"
+            Padding="0"
+            HeightRequest="56">
+      <Border.StrokeShape>
+        <RoundRectangle CornerRadius="2" />
+      </Border.StrokeShape>
+
+      <Grid VerticalOptions="FillAndExpand"
```
Comment (rthanga1) [inline Trimble.Modus.Components/Controls/TMAlert/TMAlert.xaml:35]: Why do we need 2 grids? Can't achieve the same with just one grid?

File: Trimble.Modus.Components/Images/alert_error_icon.svg
Comment (rthanga1) [inline Trimble.Modus.Components/Images/alert_error_icon.svg:1]: We are facing issues when the user adds another image with the same name. E.g  #307. Can you add a prefix 'modus' to these images? like "modus_alert_error_icon.svg". Let me know if you find any other solution for this.

**Review by rthanga1 — DISMISSED** [review]

**Review by ElishaSamPeterPrabhu — APPROVED** [review]

**Pull request merged**